src/main/java/io/neonbee/data/DataContext.java

```diff
@@ -96,6 +98,49 @@ public interface DataContext {
      */
     <T> T remove(String key);
 
+    /**
+     * Arbitrary response data of this context, which is passed backwards from callee to caller.
+     *
+     * @return all the context response data as a map.
+     */
+    Map<String, Object> responseData();
+
+    /**
+     * Returns a map between {@link DataRequest} and received data map.
+     *
+     * @return a map between {@link DataRequest} and received data map
```

Comment:
this JavaDoc could be better explaining how responseData / receivedData is to be used and also that some helper methods are available, e.g. findReceivedData, etc.